client/components/domains/transfer-domain-step/index.jsx

```diff
@@ -629,35 +629,28 @@ class TransferDomainStep extends Component {
 	getInboundTransferStatus = () => {
 		this.setState( { submittingWhois: true } );
 
-		return new Promise( ( resolve ) => {
-			checkInboundTransferStatus(
-				getFixedDomainSearch( this.state.searchQuery ),
-				( error, result ) => {
-					this.setState( { submittingWhois: false } );
-
-					if ( ! isEmpty( error ) ) {
-						resolve();
-						return;
-					}
-
-					const inboundTransferStatus = {
-						creationDate: result.creation_date,
-						email: result.admin_email,
-						loading: false,
-						losingRegistrar: result.registrar,
-						losingRegistrarIanaId: result.registrar_iana_id,
-						privacy: result.privacy,
-						termMaximumInYears: result.term_maximum_in_years,
-						transferEligibleDate: result.transfer_eligible_date,
-						transferRestrictionStatus: result.transfer_restriction_status,
-						unlocked: result.unlocked,
-					};
-
-					this.setState( { inboundTransferStatus } );
-					resolve( { inboundTransferStatus } );
-				}
-			);
-		} );
+		checkInboundTransferStatus( getFixedDomainSearch( this.state.searchQuery ) )
```

Comment:
You need to return the promise from the function, because some callers (`refreshStatus`!) use it. That's the cause of the `TypeError` you reported.

Reply:
Ah, brilliant - I was thinking of the `getInboundTransferStatus` returning inboundTransferStatus`, but that's actually the callback of the `.then` call. You're absolutely right, `return checkInboundTransferStatus( ... ) ...` resolves my `TypeError`. Thank you!